From d563f2e8d88459d0516026a521ce03073656bb90 Mon Sep 17 00:00:00 2001
From: Ivan Yonchovski <yyoncho@gmail.com>
Date: Mon, 7 Nov 2022 10:23:39 +0200
Subject: [PATCH 02/13] Add MacOS as well

---
 src/spsupr.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/spsupr.c b/src/spsupr.c
index c81478fd8d0..864c60975fd 100644
--- a/src/spsupr.c
+++ b/src/spsupr.c
@@ -1,6 +1,6 @@
 #include "spsupr.h"
 
-#if defined(__linux__) || defined(__FreeBSD__)
+#if defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
 #include <sys/types.h>
 #include <sys/socket.h>
 #include <sys/wait.h>
-- 
2.42.0

