From 6bab276350b766e3a1836d322647081a5fc19f5b Mon Sep 17 00:00:00 2001
From: Sebastian Sturm <s.sturm@arkona-technologies.de>
Date: Sun, 26 Feb 2023 20:38:00 +0100
Subject: [PATCH 09/13] Pass environment to language server

---
 src/json.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/json.c b/src/json.c
index d137c9a7501..63d9ffa553e 100644
--- a/src/json.c
+++ b/src/json.c
@@ -31,6 +31,7 @@ along with GNU Emacs.  If not, see <https://www.gnu.org/licenses/>.  */
 
 #include "lisp.h"
 #include "intervals.h"
+#include "process.h"
 #include "spsupr.h"
 #include "thread.h"
 #include "buffer.h"
@@ -1068,6 +1069,8 @@ DEFUN ("json-rpc-connection", Fjson_rpc_connection, Sjson_rpc_connection, 1,
   opts.binary = new_argv[0];
   opts.argv = new_argv;
   opts.read_timeout_ms = -1;
+  const Lisp_Object current_dir = get_current_directory (true);
+  opts.envp = make_environment_block (current_dir);
   struct SSP_Handle *handle = ssp_spawn (&opts);
   if (!handle)
     {
-- 
2.42.0

