From b328d45ce3c52c589e62a317498e460c35d96fca Mon Sep 17 00:00:00 2001
From: Sebastian Sturm <s.sturm@arkona-technologies.de>
Date: Sun, 26 Feb 2023 21:28:15 +0100
Subject: [PATCH 10/13] Use timeouts on all json-rpc handle locks

---
 src/json.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/json.c b/src/json.c
index 63d9ffa553e..c3c189a6906 100644
--- a/src/json.c
+++ b/src/json.c
@@ -1386,7 +1386,10 @@ DEFUN ("json-rpc", Fjson_rpc, Sjson_rpc, 1, MANY, NULL, doc
 	}
     }
   CALLN (Ffuncall, callback, Qnil, Qnil, Qt);
-  if (pthread_mutex_lock (&param->handle_mx) == 0)
+  // timeout probably not necessary here (?), but let's rather be safe than
+  // sorry
+  struct timespec timeout = {.tv_sec = 1, .tv_nsec = 0};
+  if (pthread_mutex_timedlock (&param->handle_mx, &timeout) == 0)
     {
       param->handle->close (param->handle);
       param->handle = NULL;
-- 
2.42.0

